### ISSUE TITLE
[Bug]:  Ipfs gateway displayed in privacy setting page is confusing when user entered an invalid value

### ISSUE THREAD

**Issue by chloeYue** [MetaMask/metamask-extension#20848]
### Describe the bug

When user enter an invalid ipfs gateway on privacy setting page, there's message shows that the input is not valid. But when reopen the setting page, the invalide gateway still displayed and truncated.

### Steps to reproduce

1. Go to privacy settings page
2. Add an invalid IPFS gateway "test test", see error message is displayed
3. Close setting page and land on home page
4. Go to privacy setting page again, see IPFS gateway is displayed as "test"

**Expected**
Either we cannot proceed with an invalid address, **or** when we reopen the settings page, the previous valid address is displayed.

https://github.com/MetaMask/metamask-extension/assets/105063779/49957d4e-4375-43e6-8f6b-0e4e39787a23

### Version

v11.1.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by danjm** [comment]
Good catch. Very interesting behaviour. I don't think this is a release blocker, but I am marking it as a sev-2. It is okay if this is fixed in a future release, imo

**Comment by darkwing** [comment]
So this works the way it does because we don't presently do a network request to validate the IPFS gateway; we simply ask for the gateway name, perpend `https` to it, and validate if the URL is correct.  With your example, the last valid address was `test` because it was valid until you added the space.

Removing bug label and changing to feature request to either add a Save button or trigger a network request to validate that the hostname exists.
